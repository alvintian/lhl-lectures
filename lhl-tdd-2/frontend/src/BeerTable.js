import React from "react";
import PropTypes from "prop-types";

const BeerTableRow = ({ name, style }) => (
  <tr>
    <td>{name}</td>
    <td>{style}</td>
  </tr>
);

const BeerTable = ({ beers }) => (
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Style</th>
        <th>ABV</th>
      </tr>
    </thead>
    <tbody>{beers.map(beer => <BeerTableRow key={beer.id} {...beer} />)}</tbody>
  </table>
);

BeerTable.propTypes = {
  beers: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      style: PropTypes.string.isRequired
    })
  ).isRequired
};
export default BeerTable;
