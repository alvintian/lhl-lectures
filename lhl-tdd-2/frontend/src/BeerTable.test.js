import React from "react";
import "./TestSetup";
import { shallow } from "enzyme";

import BeerTable from "./BeerTable";

describe("<BeerTable/>", () => {
  describe(".render()", () => {
    it("renders the proper beer table based on the props", () => {
      const props = {
        beers: [
          {
            id: 1,
            name: "Lone Pine",
            style: "American IPA"
          },
          {
            id: 2,
            name: "Lug Tread",
            style: "Kolsch"
          }
        ]
      };
      const wrapper = shallow(<BeerTable {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
