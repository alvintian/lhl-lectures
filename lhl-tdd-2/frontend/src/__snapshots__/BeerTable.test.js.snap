// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BeerTable/> .render() renders the proper beer table based on the props 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Style
      </th>
      <th>
        ABV
      </th>
    </tr>
  </thead>
  <tbody>
    <BeerTableRow
      id={1}
      key="1"
      name="Lone Pine"
      style="American IPA"
    />
    <BeerTableRow
      id={2}
      key="2"
      name="Lug Tread"
      style="Kolsch"
    />
  </tbody>
</table>
`;
